### PR TITLE
Remove jira

### PR DESCRIPTION
**Pull request by tuxing1986**
the components under 'components' directory are not touched

### CODE REVIEW

File: build-dependencies.xml [hunk 1]
```diff
@@ -63,8 +63,6 @@
 
     <property name="game_plan_service.jar" value="${direct_service_libdir}/game_plan_service.jar"/>
 
-    <property name="user_service_facade.jar" value="${direct_service_libdir}/user_service_facade.jar"/>
```
Comment (skyhit) [inline build-dependencies.xml]: @tuxing1986 it is removed, because of we built from source code?

File: conf/web/WEB-INF/applicationContext.xml [hunk 2]
```diff
@@ -794,19 +777,6 @@
           <property name="viewData.showJustForm" value="true"/>
     </bean>
 
-    <bean id="dashboardJiraIssuesReportAction"
```
Comment (skyhit) [inline conf/web/WEB-INF/applicationContext.xml]: @tuxing1986 these reports are related to JIRA calls, not direct database query?

File: build-dependencies.xml (hunk 1 above)
Comment (tuxing1986) [reply build-dependencies.xml]: yes, it's removed and the user_service_facade under the service directory is also removed.

File: docker/tc-cache/cache/deploy/jboss-web.deployer/ROOT.war/index.html [hunk 3]
```diff
@@ -19,8 +19,7 @@
    <h3>JBoss Online Resources</h3>
    <ul>
       <li><a href="http://www.jboss.org/products/jbossas/docs">JBoss Documentation</a></li>
-      <li><a href="http://wiki.jboss.org/">JBoss Wiki</a></li>
-      <li><a href="http://jira.jboss.org/">JBoss JIRA</a></li>
```
Comment (skyhit) [inline docker/tc-cache/cache/deploy/jboss-web.deployer/ROOT.war/index.html]: @tuxing1986 this is not required to remove

File: src/java/main/com/topcoder/direct/services/configs/ConfigUtils.java [hunk 4]
```diff
@@ -51,17 +51,22 @@
...
+ * @version 1.6
  */
 public final class ConfigUtils {
     /**
      * <p>
      * Overview object to hold studio overview information.
      * </P>
      */
-    private static Overview overview;
+    private static Overview overview;;
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/configs/ConfigUtils.java]: @tuxing1986 extra `;`

File: conf/web/WEB-INF/applicationContext.xml (hunk 2 above)
Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: dashboardJiraIssuesReportAction will execute "dashboard_jira_issues_report" query in the database.

File: src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java [hunk 5]
```diff
@@ -663,8 +613,6 @@ public int compare(SoftwareSubmissionDTO o1, SoftwareSubmissionDTO o2) {
             // set project bug race fee configuration
             ProjectData directProjectData = getProjectServiceFacade().getProject(currentUser,
                     softwareCompetition.getProjectHeader().getTcDirectProjectId());
-            setFixedBugRaceFee(directProjectData.getFixedBugContestFee());
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java]: @tuxing1986 is this related to JIRA ?

File: docker/tc-cache/cache/deploy/jboss-web.deployer/ROOT.war/index.html (hunk 3 above)
Comment (tuxing1986) [reply docker/tc-cache/cache/deploy/jboss-web.deployer/ROOT.war/index.html]: okay

File: src/java/main/com/topcoder/direct/services/configs/ConfigUtils.java (hunk 4 above)
Comment (tuxing1986) [reply src/java/main/com/topcoder/direct/services/configs/ConfigUtils.java]: okay

File: src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java [hunk 6]
```diff
@@ -555,9 +562,6 @@ private void setCommonProjectMetadata(List<DirectProjectMetadata> allMetadata) {
             } else if (keyId == 4L) {
                 // svn address
                 getViewData().setSvnURL(data);
-            } else if (keyId == 5L) {
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java]: @tuxing1986 this should be restored.

File: src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java (hunk 5 above)
Comment (tuxing1986) [reply src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java]: All the bug fix is related with JIRA.

File: src/java/main/com/topcoder/direct/services/view/action/project/ProjectOverviewAction.java [hunk 7]
```diff
@@ -653,10 +621,7 @@ private void setProjectGeneralInfo(ProjectData project) throws Exception {
             } else if (keyId == 4L) {
                 // svn
                 getViewData().getProjectGeneralInfo().setSvn(m.getMetadataValue());
-            } else if (keyId == 5L) {
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/action/project/ProjectOverviewAction.java]: @tuxing1986 this should be restored

File: src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java (hunk 6 above)
Comment (tuxing1986) [reply src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java]: It's related with jira things, why it should be restored?

File: src/java/main/com/topcoder/direct/services/view/action/project/edit/SaveCockpitProjectSettingAction.java [hunk 8]
```diff
@@ -267,7 +272,6 @@ protected void executeAction() throws Exception {
         metadataOperations.add(getFormData().getBudget());
         metadataOperations.add(getFormData().getDuration());
         metadataOperations.add(getFormData().getSvn());
-        metadataOperations.add(getFormData().getJira());
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/action/project/edit/SaveCockpitProjectSettingAction.java]: @tuxing1986 no need to remove this

File: src/java/main/com/topcoder/direct/services/view/action/project/planner/ProjectPlannerAction.java [hunk 9]
```diff
@@ -283,10 +283,6 @@ protected void executeAction() throws Exception {
         final ProjectData project = getProjectServiceFacade().getProject(DirectUtils.getTCSubjectFromSession(),
                 getFormData().getProjectId());
 
-        // bug race fee
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/action/project/planner/ProjectPlannerAction.java]: @tuxing1986 does this related to JIRA API calls?

File: src/web/scripts/select-a-copilot.js [hunk 10]
```diff
@@ -198,10 +202,7 @@ function handleSortSelectChangeEvent(copilotListTable) {
                 break;
             case "failureContestsInput":
                 sortIndex = 4;
-                break;
-            case "bugraceInput":
```
Comment (skyhit) [inline src/web/scripts/select-a-copilot.js]: @tuxing1986 what does this related to?

File: src/web/scripts/editCockpitProject.js [hunk 11]
```diff
@@ -1517,28 +1517,6 @@ $(document).ready(function (e) {
                 formData.svn = svnUpdate;
             }
 
-            // get project jira address
```
Comment (skyhit) [inline src/web/scripts/editCockpitProject.js]: @tuxing1986 this is not related to JIRA call?

File: conf/web/WEB-INF/applicationContext.xml (hunk 2 above)
Comment (skyhit) [reply conf/web/WEB-INF/applicationContext.xml]: @tuxing1986 can you restore the related logic, db query related logic is fine, we just need to remove the JIRA API calls, as we are going to shutdown the JIRA service.

File: src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java (hunk 5 above)
Comment (skyhit) [reply src/java/main/com/topcoder/direct/services/view/action/contest/SoftwareContestSubmissionsAction.java]: @tuxing1986 I mean this value is not related to JIRA API calls, it is just general settings for project, right?  if yes, I think we can keep it, it is not harmful, and useful for auditing.

File: src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java (hunk 6 above)
Comment (skyhit) [reply src/java/main/com/topcoder/direct/services/view/action/project/edit/EditCockpitProjectAction.java]: @tuxing1986 is not it just related to set the project meta info for Bug Trace, I mean an URL?

File: conf/web/WEB-INF/applicationContext.xml (hunk 2 above)
Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: dashboardJiraIssuesReportAction will also use  ServerConfiguration.JIRA_SERVER_NAME indirectly.
Hence, it will invoke the jira calls.

File: src/web/scripts/select-a-copilot.js (hunk 10 above)
Comment (tuxing1986) [reply src/web/scripts/select-a-copilot.js]: All the bugraces are related to jiras as they are post as jira issues.

File: conf/web/WEB-INF/applicationContext.xml (hunk 2 above)
Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: dashboardJiraIssuesReportAction is for src\web\WEB-INF\report\dashboard-jira-issues-report.jsp
That jsp page will call '<a href="https://<%=ServerConfiguration.JIRA_SERVER_NAME%>/browse/${ticketId}">${ticketId}</a>' to get the ticket.

Hence all the action, jsp file and the related js file should all be removed.

Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: call https:/ServerConfiguration.JIRA_SERVER_NAM/browse/ticketId

Comment (skyhit) [reply conf/web/WEB-INF/applicationContext.xml]: @tuxing1986 this seems just linked to the external service, let me double check with PM, and response to you

Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: okay, all the issues and bug races related to the jira urls, hence they should be removed.

Comment (tuxing1986) [reply conf/web/WEB-INF/applicationContext.xml]: I can not access my gmail now, please post any issue here, thanks.
